File: app/src/main/java/com/ttd/jigsawpuzzlev1/ui/BaseActivity.java
package com.ttd.jigsawpuzzlev1.ui;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.ttd.jigsawpuzzlev1.utils.PermissionsUtil;

public class BaseActivity extends AppCompatActivity {
    protected PermissionsUtil permissionsUtil;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        initUtils();
    }

    private void initUtils() {
        permissionsUtil = new PermissionsUtil(this);
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        permissionsUtil.disposePermissionResult(requestCode, permissions, grantResults);
    }
}

File: app/src/main/java/com/ttd/jipu/ui/BaseActivity.kt
package com.ttd.jipu.ui

import android.annotation.SuppressLint
import android.content.pm.ActivityInfo
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

@SuppressLint("Registered")
open class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED

    }
}
